limits_definition.py

```diff
@@ -0,0 +1,26 @@
+Parameter_count = 3 
+Parameter_name =  ("Temparature","SOC","Charge Rate")
+LOW = 0
+HIGH = 1
+
+#Extension 1: Early Warning
+# Introducing a 'warning' level with a tolerance of 5% of the upper-limit.
+# Temparature : 5% of 45 = 2.25
+# SOC  : 5% of 80 = 4
+# Charge rate : 5% of 0.8 = 0.04
+MIN_TEMP = 0
+MAX_TEMP = 45
+MIN_SOC = 20
+MAX_SOC = 80
+MIN_CHARGE_RATE = 0.1 #assumption
+MAX_CHARGE_RATE = 0.8
+MIN_WARNING_TEMP = 2.25 
+MIN_WARNING_SOC = 24 
+MIN_WARNING_CHARGE_RATE = 0.14
+MAX_WARNING_TEMP = 42.75
+MAX_WARNING_SOC = 76
+MAX_WARNING_CHARGE_RATE = 0.76
+Low_Breach_values = MIN_TEMP, MIN_SOC , MIN_CHARGE_RATE

```

Comment:
Maintaining arrays is very cumbersome from a maintenance point of view.
One parameter addition requires changes in quite few places. 
There is scope for improvement from the complexity handling point of view.
Also readability of code takes a big hit, which is a crucial factor in maintaining  code in future.